pkg/urbit/jets/e/pedersen.c

```diff
@@ -0,0 +1,100 @@
+
+/* j/5/pedersen.c
+**
+*/
+#include "all.h"
+#include <urcrypt.h>
+
+static c3_y*
+u3r_bytes_all_works(c3_w* len_w, u3_atom a)
+{
+  c3_w met_w;
+  if (a == u3_nul) {
+    met_w = *len_w = 1;
+  } else {
+    met_w = *len_w = u3r_met(3, a);
+  }
+  return u3r_bytes_alloc(0, met_w, a);
+}
+
+static c3_y *
+fold_hash(u3_atom x, c3_w *len_acc) {
+  c3_w len_x, pos = 0, ext_pos = 0;
+  c3_y *acc = NULL, *out = NULL, *ext = NULL;
```

Comment:
these should be static buffers, there is no reason to involve malloc here is there?

Reply:
by static i mean stack allocated, a la

    c3_y acc_y[32];